object_store/src/aws/mod.rs

```diff
@@ -319,7 +320,7 @@ impl ObjectStore for AmazonS3 {
 
 #[derive(Debug)]
 struct S3MultiPartUpload {
-    part_idx: usize,
```

Comment:
Not sure I understand this change. Using an atomic here would be irrelevant as in both `put_part` and `complete` we have a mutable reference

Reply:
yeah, you're right 🤦